task_list/app/controllers/tasks_controller.rb

```diff
@@ -26,8 +27,24 @@ def destroy
     redirect_to '/'
   end
 
+  def edit
+    @task = Task.find(params[:id])
+    @title = "Edit Task"
+    @action = "update"
+  end
+
+  def toggle_complete
+    Task.update(params[:id], :completed_at => Time.now)
+    redirect_to '/'
+  end
+
+  def update
+    Task.update(params[:id], name: task_params[:task][:name], description: task_params[:task][:description])
```

Comment:
Because you have the `task_params` method I would recommend simplifying this call to `Task.update(params[:id], task_params[:task])`.